pkg/metrics/service-monitor.go

```diff
@@ -15,12 +15,45 @@
 package metrics
 
 import (
-	monitoringv1 "github.com/coreos/prometheus-operator/pkg/client/monitoring/v1"
+	"errors"
+
+	monitoringv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	monclientv1 "github.com/coreos/prometheus-operator/pkg/client/versioned/typed/monitoring/v1"
+	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
 )
 
-// GenereateServiceMonitor generates a prometheus-operator ServiceMonitor object
+// CreateServiceMonitors creates ServiceMonitors objects based on an array of Service objects.
+// If CR ServiceMonitor is not registered in the Cluster it will not attempt at creating resources.
+func CreateServiceMonitors(config *rest.Config, ns string, services []*v1.Service) ([]*monitoringv1.ServiceMonitor, error) {
+	// check if we can even create ServiceMonitors
+	exists, err := IsServiceMonitor(config)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, errors.New("creating ServiceMonitors failed as object is not registered in the API")
+	}
+
+	var serviceMonitors []*monitoringv1.ServiceMonitor
+	mclient := monclientv1.NewForConfigOrDie(config)
+
+	for _, s := range services {
+		sm := GenerateServiceMonitor(s)
+		smc, err := mclient.ServiceMonitors(ns).Create(sm)
+		if err != nil {
+			return serviceMonitors, err
```

Comment:
```suggestion
			return []*monitoringv1.ServiceMonitor{}, err
```

Reply:
The `serviceMonitors` might contain some already created objects, so I think returning that makes sense.